### PR TITLE
Example showing how to configure an aggregate-realm with a principal-…

### PR DESCRIPTION
**Pull request by ashley-abdelsayed98**
…transformer

This example depends on the changes in wildfly/wildfly-core#3825

### CODE REVIEW

File: aggregate-realm-principal-transformer/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+## Using a principal-transformer in aggregate-realm
+
+This example shows how to configure an ```aggregate-realm``` with a ```principal-transformer```. We will create two filesystem 
+realms to be used in aggregate in our ```aggregate-realm```: one will hold the credentials for authentication of identity
```
Comment (fjuma) [inline aggregate-realm-principal-transformer/README.md]: "authentication of identity" -> "authentication of the identity"

File: aggregate-realm-principal-transformer/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+The following command can now be used to configure the ```aggregate-realm``` and secure the web application
+with an HTTP authentication mechanism which will use our aggregate-realm.
+
+```
+    $SERVER_HOME/bin/jboss-cli.sh --connect --file=$PATH_TO_ELYTRON_EXAMPLES/aggregate-realm-principal-transformer/configure-elytron-server.cli
+```
+
+It creates a new identity with the username: "guest" and password: "guestPwd1!". It also adds the attribute Roles=Users to the 
+identity as that role is required to be authorized to access the application. The password is stored in the authorization realm 
+under the name "guest" and the attribute is stored in the authentication realm under the name "guest-attributes" and so
```
Comment (fjuma) [inline aggregate-realm-principal-transformer/README.md]: I think this sentence has things flipped around :) The password is stored in the authentication realm and the role is stored in the authorization realm.

File: aggregate-realm-principal-transformer/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+```
+cd $PATH_TO_ELYTRON_EXAMPLES/simple-webapp
+mvn clean install wildfly:deploy
+```
+
+You can access the application at localhost:8080/simple-webapp. 
+Try logging in with the username: "guest" and password: "guestPwd1!"
+You are successfully authorized to access the secured application, which means our transformer worked and the attribute
+Roles=Users was successfully loaded from the ```authorization-realm``` using the transformed principal.
```
Comment (fjuma) [inline aggregate-realm-principal-transformer/README.md]: Maybe you could mention here that the user can also try removing the principal-transformer from the aggregate-realm configuration to see that without it, they won't be able to access the application.

File: aggregate-realm-principal-transformer/configure-elytron-server.cli [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#Configure a regular expression transformer
+/subsystem=elytron/regex-principal-transformer=RegexTransformer:add(pattern="guest", replacement="guest-attributes")
```
Comment (fjuma) [inline aggregate-realm-principal-transformer/configure-elytron-server.cli]: Very minor comment on style, we usually use a lower case character for the first character in a resource name:

RegexTransformer -> regexTransformer

**Comment by fjuma** [comment]
Nice work, @ashley-abdelsayed98! I think it would be good to add a comment in the description of this PR to note that it depends on your WildFly Core PR. Thanks!

File: aggregate-realm-principal-transformer/README.md (hunk 1 above)
Comment (ashley-abdelsayed98) [reply aggregate-realm-principal-transformer/README.md]: Fixed

File: aggregate-realm-principal-transformer/README.md (hunk 2 above)
Comment (ashley-abdelsayed98) [reply aggregate-realm-principal-transformer/README.md]: Fixed

File: aggregate-realm-principal-transformer/README.md (hunk 3 above)
Comment (ashley-abdelsayed98) [reply aggregate-realm-principal-transformer/README.md]: Added

File: aggregate-realm-principal-transformer/configure-elytron-server.cli (hunk 4 above)
Comment (ashley-abdelsayed98) [reply aggregate-realm-principal-transformer/configure-elytron-server.cli]: fixed

**Comment by fjuma** [comment]
Thanks @ashley-abdelsayed98!

**Pull request merged**